src/CommunityPatch/Patches/Perks/Vigor/TwoHanded/EvisceratorPatch.cs

```diff
@@ -0,0 +1,117 @@
+
+    public override IEnumerable<MethodBase> GetMethodsChecked() {
+      yield return TargetMethodInfo;
+    }
+
+    private PerkObject _perk;
+
+    private static readonly byte[][] Hashes = {
+      new byte[] {
+        // e1.1.0.225190
+        0xDC, 0x73, 0xE5, 0x9D, 0x02, 0x06, 0x68, 0x6B,
+        0x9F, 0x87, 0xE2, 0xD1, 0xE4, 0x96, 0x9E, 0xB7,
+        0xA5, 0xA1, 0x21, 0xCA, 0x9E, 0xC5, 0x06, 0xE9,
+        0xB2, 0xD3, 0x58, 0x35, 0x5B, 0x27, 0x14, 0x72
+      }
+    };
+
+    public override void Reset()
+      => _perk = PerkObject.FindFirst(x => x.Name.GetID() == "C2CwsC91");
+
+    // ReSharper disable once CompareOfFloatsByEqualityOperator
+    public override bool? IsApplicable(Game game) {
+      if (_perk == null) return false;
+      if (_perk.PrimaryBonus != 0.3f) return false;
+
+      var patchInfo = Harmony.GetPatchInfo(TargetMethodInfo);
+      if (AlreadyPatchedByOthers(patchInfo)) return false;
+
+      var hash = TargetMethodInfo.MakeCilSignatureSha256();
+      return hash.MatchesAnySha256(Hashes);
+    }
+
+    public override void Apply(Game game) {
+      var textObjStrings = TextObject.ConvertToStringList(
+        new List<TextObject> {
+          _perk.Name,
+          _perk.Description
+        }
+      );
+      // most of the properties of skills have private setters, yet Initialize is public
+      _perk.Initialize(
+        textObjStrings[0],
+        textObjStrings[1],
+        _perk.Skill,
+        (int) _perk.RequiredSkillValue,
+        _perk.AlternativePerk,
+        _perk.PrimaryRole, -30f,
+        _perk.SecondaryRole, _perk.SecondaryBonus,
+        SkillEffect.EffectIncrementType.AddFactor
+      );
+      if (Applied) return;
+
+      CommunityPatchSubModule.Harmony.Patch(TargetMethodInfo, new HarmonyMethod(PatchMethodInfo));
+      Applied = true;
+    }
+
+    // ReSharper disable once InconsistentNaming
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    public static bool Prefix(AgentMoraleInteractionLogic __instance, Agent affectedAgent, Agent affectorAgent, AgentState agentState) {
+      var affectorCharacter = (CharacterObject) affectorAgent?.Character;
+      
+      if (affectorCharacter == null) return false;
+      if (affectedAgent.Character == null) return false;
+      if (agentState != AgentState.Killed && agentState != AgentState.Unconscious) return false;
+      if (affectedAgent.Team == null) return false;
+      
+      var moralChangesTuple = MissionGameModels.Current.BattleMoraleModel.CalculateMoraleChangeAfterAgentKilled(affectedAgent);
+      var moralChangeFriend = moralChangesTuple.Item1;
+      var moralChangeEnemy = moralChangesTuple.Item2;
+      
+      if (!AnyMoralChanges(moralChangeEnemy, moralChangeFriend)) return false;
+
+      if (affectorCharacter.GetPerkValue(ActivePatch._perk)) 
+        moralChangeFriend += moralChangeFriend * ActivePatch._perk.PrimaryBonus / -100;
```

Comment:
This follows the pattern of factors they use elsewhere.

Reply:
I haven't found this pattern yet. What I noticed, for the eviscerator perk specifically, is that TW's implementation don't work with the perk bonus being in this '0.3' format. Take a look in the method SandboxAgentApplyDamageModel.CalculateMoraleEffects.

The method expects the perk to have the value '30' (not 0.3) to apply a morale buff of 30%. It instantiates the explainedNumber with a morale value (usually 50), then call AddPerkBonusCharacter as it is. This would causes the morale to be increased by 0.03% if perk was .3, erroneously.
```
PerkHelper.AddPerkBonusForCharacter(eviscerator, characterObject, ref explainedNumber);
```